mimic/core.py

```diff
@@ -102,25 +114,51 @@ def add_api(self, api):
                 " does not implement IAPIMock or IExternalAPIMock"
             )
 
-    def get_external_api(self, api_name):
+    def remove_external_api(self, api_id):
+        """
+        Remove an External API from the listing.
+
+        :param text_type api_id: the id of the API instance
+            e.g 3845-39583, cloudfiles-mimic
+        """
+        if api_id in self._uuid_to_api_external:
+            api = self._uuid_to_api_external[api_id]
+
+            if len(api.list_templates()) == 0:
+                del self._uuid_to_api_external[api_id]
+
```

Comment:
style nitpick: this seems like a lot of vertical whitespace.  Especially given that these diffs are kinda big to begin with, would you mind compressing a little?  I don't mind a line or two separating different sections in a function, but after each "`else:`" is a bit much.